### ISSUE TITLE
Specified method is not supported

### ISSUE THREAD

**Issue by manidula** [replaysMike/AnyDiff#4]
`
System.NotSupportedException: Specified method is not supported.
   at Newtonsoft.Json.Linq.JProperty.JPropertyList.<GetEnumerator>d__1.System.Collections.IEnumerator.Reset()
   at ObjectCompare.DiffProvider.GetCountFromEnumerable(IEnumerable enumerable)
   at ObjectCompare.DiffProvider.GetDifferences(String propertyName, Type propertyType, TypeConverter typeConverter, Object left, Object right, Object parent, ICollection`1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection`1 propertyList)
   at ObjectCompare.DiffProvider.RecurseProperties(Object left, Object right, Object parent, ICollection`1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection`1 propertyList)
   at ObjectCompare.DiffProvider.GetDifferences(String propertyName, Type propertyType, TypeConverter typeConverter, Object left, Object right, Object parent, ICollection`1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection`1 propertyList)
   at ObjectCompare.DiffProvider.RecurseProperties(Object left, Object right, Object parent, ICollection`1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection`1 propertyList)
`

The issue is happening when you have different objects inherited from one parent object and we put them in the list (List<Parent>) container. 

        private static long GetCountFromEnumerable(IEnumerable enumerable)
        {
            if (enumerable == null)
                return 0L;
            var count = 0L;
            var enumerableType = enumerable.GetType().GetExtendedType(DefaultTypeSupportOptions);
            if (enumerableType.IsCollection)
                return ((ICollection)enumerable).Count;
            if (enumerableType.IsArray)
                return ((Array)enumerable).LongLength;
           if (enumerableType.IsEnumerable)
            {
                if (enumerable is IEnumerable<Newtonsoft.Json.Linq.JToken> jPropertyList)
                {
                    return jPropertyList.Count();  // I added this line to fix the issue
                }
            }

            // count the enumerable
            foreach (var row in enumerable)
                count++;

            // try to reset the enumerable, if IEnumerable implements it
            var enumerator = enumerable.GetEnumerator();
            try
            {
                enumerator.Reset();
            }
            catch (NotImplementedException) { }
            return count;
        }

**Comment by replaysMike** [comment]
Thank you for the report, I will prioritize this

**Comment by replaysMike** [comment]
This is now included as of release version 1.0.61

**Comment by replaysMike** [comment]
This is tested with the following, let me know if you have any issues with it.

```csharp
var obj1 = JToken.Parse(@"{ ""list"": [""test1"", ""test2""] }");
var obj2 = JToken.Parse(@"{ ""list"": [""test3"", ""test4""] }");
var diff = AnyDiff.Diff(obj1, obj2);
Assert.AreEqual(4, diff.Count);
```
